log4j-api/src/main/java/org/apache/logging/log4j/internal/DefaultLogBuilder.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.internal;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.SimpleMessage;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.apache.logging.log4j.util.StackLocatorUtil;
+import org.apache.logging.log4j.util.Supplier;
+
+
+/**
+ * Collects data for a log event and then logs it.
+ */
+public class DefaultLogBuilder implements LogBuilder {
+    private static Message EMPTY_MESSAGE = new SimpleMessage("");
+    private static final String FQCN = DefaultLogBuilder.class.getName();
+    private static final Logger LOGGER = StatusLogger.getLogger();
+
+    private final Logger logger;
+    private Level level;
+    private Marker marker;
+    private Throwable throwable;
+    private StackTraceElement location;
+    private Object object;
+    private Message msg;
+    private String textMessage;
+    private Supplier<Message> supplier;
+    private Object[] parameters;
```

Comment:
The current implementation retains references parameter values between fluent logger uses, this can cause objects that are expected to be short lived to make their way into tenured space.

Reply:
These have all been removed with the changes Remko requested. I don't believe I have committed them yet.